java/code/src/com/redhat/rhn/taskomatic/task/threaded/QueueDriver.java

```diff
@@ -67,4 +67,10 @@
      * Actions that has to be executed, when queue is created
      */
     void initialize();
+
+    /**
+     * Specify if this is a blocking worker thread
+     * @return true if the task queue should wait end of the workers before continue
+     */
+    boolean isBlockingTaskQueue();
```

Comment:
```suggestion
    default boolean isBlockingTaskQueue() {
        return false;
    }
```

This is looks like a good candidate for having an interface default method that returns `false`. In this way you can avoid adding it explicitly to the other driver implementations.